### ISSUE TITLE
Using L.seed_everything in main() yields wrong labels when using shuffle in Dataloader

### ISSUE THREAD

**Issue by RobinBram** [Lightning-AI/pytorch-lightning#19619]
### Bug description

Using L.seed_everything in main() yields wrong labels when using shuffle in Dataloader.

### What version are you seeing the problem on?

v2.2

### Error messages and logs

```
# Error messages and logs here please
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>

**Comment by RobinBram** [comment]
Completely removed seeding but having shuffle = True in Dataloader still yields wrong labels.

**Comment by RobinBram** [comment]
Some else is bonkers with my code. NVM
